CRM/Batch/Form/Entry.php

```diff
@@ -838,7 +831,7 @@ private function processMembership(&$params) {
             $value['contact_id'], $value['membership_type_id'], FALSE,
             //$numTerms should be default to 1.
             NULL, NULL, $value['custom'], 1, NULL, FALSE,
-            NULL, $membershipSource, $isPayLater, $campaignId, $formDates
+            NULL, $membershipSource, $isPayLater, ['campaign_id' => $value['campaign_id'] ?? NULL], $formDates
```

Comment:
The element build on the batch entry form is `member_campaign_id` so think we should replace the key in `$value` with the same? The patch after being applied is still not able to update the campaign in the membership due to the above mismatch. Replacing it with

    NULL, $membershipSource, $isPayLater, ['campaign_id' => $value['member_campaign_id'] ?? NULL], $formDates

works fine. @eileenmcnaughton Can you confirm pls?

Reply:
thanks @jitendrapurohit I've updated it as you suggest